### ISSUE TITLE
Cache data in order to speed up data retrieval

### ISSUE THREAD

**Issue by adibbehjat** [dancingcactus/python-omniture#35]
I've noticed that every time I run Python Omniture, it conducts the expensive request every time by:

- Collecting the latest report suites from file
- Get Metrics
- Get Elements
- Get Segment

Instead of submitting these requests every time, it would be best if there is a module or command that can update and access a group of files - for example, named "suite.json", "metrics.json", "elements.json", and "segments.json" - in order to reduce the number of unnecessary call requests.

I understand some update their metrics or add additional elements, but this can be handled by an 'update_file' module in order to ensure efficiency

**Comment by dancingcactus** [comment]
Great idea. I like think that would save a bunch of time between executions. I wonder if we could add this as an option when you initialize the object. Something like `omniture.authenticate(username, secrect, cached=True)`  would that work for what your use cases? If we did this I think we would have to have a function to update the cache.

**Comment by adibbehjat** [comment]
That's a good idea! Just to give an idea, my use case looks like something like this:
I utilize a heavy for loop to conduct bulk searches, where I would like to identify the trend/behavior of users of specific device arriving from a specific campaign. Therefore, the pseudo code would be something like

```python
list_of_campaigns = [...]
list_of_devices = [...]
queue = []

for campaign in list_of_campaigns:
    for device in list_of_devices:
        # Build report
        # Append to queue

# Send request to Adobe
```

Based on O notation, once the campaign list expands, it takes longer and longer to process the data, and I've noticed that it does it every time for any report I build. To have a cache would make the program run effectively.

A better implementation could be to have the cache be initiated at the very start of the request when I establish communication, cache the results in json files once and only once, and then check if file exists and utilize that. In future requests, whenever one runs a new request, the cache file will be re-written with the latest metric, element, or segment.

**Comment by dancingcactus** [comment]
This is now solved by #38
